doc/nimc.rst

```diff
@@ -325,6 +325,46 @@ Debugger option
 The ``debugger`` option enables or disables the *Embedded Nim Debugger*.
 See the documentation of endb_ for further information.
 
+Code hot reloading
+------------------
+**Note:** At the moment code hot reloading is supported only in
+JavaScript projects.
+
+The `hotReloading` option enables special compilation mode where changes in
+the code can be applied automatically to a running program. The code reloading
+happens at the granularity of an individual module. When a module is reloaded,
+Nim will preserve the state of all global variables which are initialized with
+a standard variable declaration in the code. All other top level code will be
+executed repeatedly on each reload. If you want to prevent this behavior, you
+can guard a block of code with the `once` construct:
+
+.. code-block:: Nim
+  var settings = initTable[string, string]()
+
+  once:
```

Comment:
Wouldn't it make more sense to do it the other way around? It seems like most code will have to be under this ``once`` invokation, but maybe I'm wrong.

Reply:
I think you are wrong, but we'll see. We already use the hot reloading branch in a quite complex project and so far this policy makes more sense.